pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go

```diff
@@ -129,10 +129,12 @@ func (e *AzureMonitorDatasource) buildQueries(queries []backend.DataQuery, dsInf
 			dimSB.WriteString(fmt.Sprintf("%s eq '%s'", dimension, dimensionFilter))
 		} else {
 			for i, filter := range azJSONModel.DimensionFilters {
-				if filter.Operator != "eq" && filter.Filter == "*" {
+				if filter.Operator != "eq" && filter.Filter == "*" && len(filter.Filters) == 0 {
```

Comment:
`Filters` doesn't get cleared when it switches to `starts with`, so it gets a `400 Bad Request` if you:
1) Set a `eq` operation with 2 values
2) switch to the `starts with` operation

Reply:
Great catch here, have fixed this 😊